common/src/test/java/tachyon/replay/ReplayCacheTest.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.replay;
+
+import java.io.IOException;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.base.Throwables;
+
+import tachyon.exception.FileAlreadyExistsException;
+import tachyon.exception.TachyonException;
+import tachyon.exception.TachyonExceptionType;
+import tachyon.replay.ReplayCache.ReplayCallable;
+import tachyon.replay.ReplayCache.ReplayCallableThrowsIOException;
+import tachyon.thrift.TachyonTException;
+import tachyon.thrift.ThriftIOException;
+
+/**
+ * Tests for {@link ReplayCache}.
+ */
+public final class ReplayCacheTest {
+  private static final Long MAX_SIZE = 10L;
+  private static final Long TIMEOUT_MS = 1000L;
+  private static final String ERROR_MESSAGE = "test error message in ReplayCacheTest";
+
+  private ReplayCache<Long> mCache;
+
+  @Before
+  public void before() {
+    mCache = ReplayCache.newReplayCache(MAX_SIZE, TIMEOUT_MS);
+  }
+
+  /**
+   * Tests that the {@link ReplayCallable} is only called when the key doesn't exist in the cache.
+   */
+  @Test
+  public void testCaching1() throws Exception {
+    CounterCallable counter = new CounterCallable();
+    Assert.assertEquals(1L, (long) mCache.run("key1", counter));
+    // Should return the first value for key1
+    Assert.assertEquals(1L, (long) mCache.run("key1", counter));
```

Comment:
this test is potentially flaky if more the gap between the subsequent calls with `key`` is more than 1 second ... is is possible to use an unlimited expiry for the cache for the sake of test to avoid this?